src/transformers/models/tapas/modeling_tapas.py

```diff
@@ -0,0 +1,2317 @@
+
+        self.number_of_token_type_embeddings = len(config.type_vocab_sizes)
+
+        # self.LayerNorm is not snake-cased to stick with TensorFlow model variable name and be able to load
+        # any TensorFlow checkpoint file
+        self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+        self.config = config
+
+    def forward(self, input_ids=None, token_type_ids=None, position_ids=None, inputs_embeds=None):
+        if input_ids is not None:
+            input_shape = input_ids.size()
+        else:
+            input_shape = inputs_embeds.size()[:-1]
+
+        seq_length = input_shape[1]
+        device = input_ids.device if input_ids is not None else inputs_embeds.device
+
+        if position_ids is None:
+            # create absolute position embeddings
+            position_ids = torch.arange(seq_length, dtype=torch.long, device=device)
+            position_ids = position_ids.unsqueeze(0).expand(input_shape)
+            # when self.config.reset_position_index_per_cell is set to True, create relative position embeddings
+            if self.config.reset_position_index_per_cell:
+                col_index = IndexMap(
+                    token_type_ids[:, :, 1], self.config.type_vocab_sizes[1], batch_dims=1
+                )  # shape (batch_size, seq_len)
+                row_index = IndexMap(
+                    token_type_ids[:, :, 2], self.config.type_vocab_sizes[2], batch_dims=1
+                )  # shape (batch_size, seq_len)
+                full_index = ProductIndexMap(col_index, row_index)  # shape (batch_size, seq_len)
+
+                first_position_per_segment = reduce_min(position_ids, full_index)[
+                    0
+                ]  # shape (max_rows * max_columns,). First absolute position for every cell
+                first_position = gather(
+                    first_position_per_segment, full_index
+                )  # ? shape (batch_size, seq_len). First absolute position of the cell for every token
+                position = torch.arange(seq_length, dtype=torch.long, device=device).unsqueeze(0)  # shape (1, seq_len)
+                position_ids = torch.min(
+                    torch.as_tensor(self.config.max_position_embeddings - 1, device=device), position - first_position
+                )
+
+        if token_type_ids is None:
+            token_type_ids = torch.zeros(
+                (input_shape + self.number_of_token_type_embeddings), dtype=torch.long, device=device
+            )
+
+        if inputs_embeds is None:
+            inputs_embeds = self.word_embeddings(input_ids)
+
+        position_embeddings = self.position_embeddings(position_ids)
+
+        embeddings = inputs_embeds + position_embeddings
+
+        token_type_embedding_name = "token_type_embeddings"
+
+        for i in range(self.number_of_token_type_embeddings):
+            name = f"{token_type_embedding_name}_{i}"
```

Comment:
```suggestion
            name = f"token_type_embeddings_{i}"
```

Reply:
This should be `token_type_embedding_name` rather than `token_type_embeddings_`, otherwise it will break.

Reply:
Fixed this, indeed cleaner.